wizard_agent/brain/treasure_strategy.py

```diff
@@ -20,20 +20,47 @@ def _get_nearest_treasure(location, treasure_list):
     else:
         return None
 
+def _get_furthest_treasure_from_opponent(opponent_location, treasure_list):
+    if treasure_list and len(treasure_list) > 1:
+        treasure_distance = 10
+        furthest_treasure = treasure_list[0]
+        for treasure in treasure_list:
+            new_treasure_dist = utils.manhattan_distance(opponent_location, treasure)
+            if new_treasure_dist > treasure_distance:
+                treasure_distance = new_treasure_dist
+                furthest_treasure = treasure
+        return furthest_treasure
+    else:
+        return None
 
 class TreasureStrategy(strategy.Strategy):
     def execute(self, game_state: object, player_state: object) -> List[str]:
         location = player_state.location
         treasure = game_state.treasure
+        list_of_opponents = game_state.opponents(player_state.id)
+        opponent_location = 0
+        for opponent in list_of_opponents:
+            opponent_location = opponent
         treasures = utils.get_reachable_tiles(location, treasure, game_state)
         # get the nearest treasure to the player
         nearest_treasure = _get_nearest_treasure(location, treasures)
+
+        # get the furthest treasure from the opponent player
+        furthest_treasure_from_opponent = _get_furthest_treasure_from_opponent(opponent_location, treasures)
+
         # navigate to the treasure
         if nearest_treasure is not None:
-            path = utils.get_shortest_path(location, nearest_treasure, game_state)
-            action_seq = utils.get_path_action_seq(location, path)
-            return action_seq
-
+            if utils.isOpponentCloser(location, opponent_location, nearest_treasure) is False:
+                print("I am closer, Yay!")
+                path = utils.get_shortest_path(location, nearest_treasure, game_state)
+                action_seq = utils.get_path_action_seq(location, path)
+                return action_seq
+            else:
+                print("I was not closer! I'm going to find another treasure that's far away from my opponent!")
+                if furthest_treasure_from_opponent is not None:
+                    path = utils.get_shortest_path(location, furthest_treasure_from_opponent, game_state)
+                    action_seq = utils.get_path_action_seq(location, path)
+                    return action_seq
         return [constants.ACTIONS["none"]]
 
     def can_execute(self, game_state: object, player_state: object) -> bool:
```

Comment:
This strategy can only execute if there's a treasure that's far from opponent or there's a reachable treasure that's furthest from opponent

Reply:
What else could we do if not? Try find something else?